### PR TITLE
Auto upsert-study

### PR DESCRIPTION
**Pull request by kim0**
Resolves https://github.com/brave/devops/issues/8609

### CODE REVIEW

File: .github/ci/griffin-study-utils.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+import json
```
Comment (mihaiplesa) [inline .github/ci/griffin-study-utils.py:2]: Is there usually a newline between line 1 and 2 here?

File: .github/workflows/upsert-study.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        required: true
+        default: "RELEASE,BETA,NIGHTLY"
+        type: string
+      platform:
+        description: "Platforms to enable on, in comma separated value format"
+        required: true
+        default: "WINDOWS,MAC,LINUX,ANDROID"
+        type: string
+      min_version:
+        description: "Minimum version to enable on [Optional]"
```
Comment (mihaiplesa) [inline .github/workflows/upsert-study.yml:28]: `(optional)`

File: .github/workflows/upsert-study.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        type: string
+      channel:
+        description: "Channels to enable on, in comma separated value format"
+        required: true
+        default: "RELEASE,BETA,NIGHTLY"
+        type: string
+      platform:
+        description: "Platforms to enable on, in comma separated value format"
+        required: true
+        default: "WINDOWS,MAC,LINUX,ANDROID"
```
Comment (mihaiplesa) [inline .github/workflows/upsert-study.yml:25]: Sort?

File: .github/workflows/upsert-study.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        description: "Feature name to enable"
+        required: true
+        type: string
+      probability_enabled:
+        description: "Percentage number to enable. ex: 25"
+        required: true
+        default: "25"
+        type: string
+      channel:
+        description: "Channels to enable on, in comma separated value format"
```
Comment (mihaiplesa) [inline .github/workflows/upsert-study.yml:18]: `(in comma separated value format)`

File: .github/workflows/upsert-study.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      name:
+        description: "Study name"
+        required: true
+        type: string
+      enable_feature:
+        description: "Feature name to enable"
+        required: true
+        type: string
+      probability_enabled:
+        description: "Percentage number to enable. ex: 25"
```
Comment (mihaiplesa) [inline .github/workflows/upsert-study.yml:13]: `(ex: 25)`

**Review by mihaiplesa — CHANGES_REQUESTED** [review]

File: .github/ci/griffin-study-utils.py (hunk 1 above)
Comment (kim0) [reply .github/ci/griffin-study-utils.py:2]: `autopep8` is not adding any, and linter is happy. So I don't think so

File: .github/workflows/upsert-study.yml (hunk 2 above)
Comment (kim0) [reply .github/workflows/upsert-study.yml:28]: Done

File: .github/workflows/upsert-study.yml (hunk 3 above)
Comment (kim0) [reply .github/workflows/upsert-study.yml:25]: Don't think we should here. The json file contains this particular order everywhere, and I think it's better to add the new ones in the same order

**Review by mihaiplesa — APPROVED** [review]

**Comment by kim0** [comment]
@antonok-edm can you please review this PR. This was done as a Github Action, since we're generally moving away from Jenkins. Also I had to reformat the `seed.json` file since it gets parsed and rewritten every time. So to get good diffs, the order and formatting needs to be deterministic. I hope this will be OK.

I have used this command to auto-format `python3 -m json.tool --sort-keys --indent 4 seed/seed.json seed/seed.json.out && mv seed/seed.json.out seed/seed.json` bec it is compatible with the output from the `.github/ci/griffin-study-utils.py` script.

Here are two sample output PRs #448 and #447 .. They contain unneeded files, just bec this PR is not yet merged. Once merged, they should only contain changes from `seed/seed.json`

Also, here's how the slack notification message looks like
slack-notification-griffin

Since GHA does not allow multi-select choice boxes, I ended up using comma separated values like `WINDOWS,MAC,LINUX`. I think we have to get this PR merged before we can see the input UI. Let me know if any changes is needed!

**Review by antonok-edm — APPROVED** [review]
This is really neat, great work @kim0 !

**Review by mihaiplesa — APPROVED** [review]

**Pull request merged**